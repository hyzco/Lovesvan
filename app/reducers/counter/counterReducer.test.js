import counterReducer from './counterReducer';

describe('CounterReducer', () => {
  describe('counter', () => {
    it('Should provide the initial state', () => {
      let initialState = { counter: 0 };
      expect(counterReducer(undefined, {})).toMatchSnapshot(initialState);
    });

    it('Should handle INCREMENT action', () => {
      let state = { counter: 1 };
      expect(counterReducer(state, { type: 'INCREMENT' })).toMatchSnapshot({
        counter: 2
      });
    });

    it('Should handle DECREMENT action', () => {
      let state = { counter: 1 };
      expect(counterReducer(state, { type: 'DECREMENT' })).toMatchSnapshot({
        counter: 0
      });
    });

    it('Should handle unknown action', () => {
      let initialState = { counter: 0 };
      expect(counterReducer(initialState, { type: 'unknown' })).toMatchSnapshot({ counter: 0 });
    });
  });
});
