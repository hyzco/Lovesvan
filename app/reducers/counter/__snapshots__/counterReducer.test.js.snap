// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterReducer counter Should handle DECREMENT action 1`] = `
Object {
  "counter": 0,
}
`;

exports[`CounterReducer counter Should handle INCREMENT action 1`] = `
Object {
  "counter": 2,
}
`;

exports[`CounterReducer counter Should handle unknown action 1`] = `
Object {
  "counter": 0,
}
`;

exports[`CounterReducer counter Should provide the initial state 1`] = `
Object {
  "counter": 0,
}
`;
